### DISCUSSION TITLE
Automatically Installing Dependencies from Workflow

### DISCUSSION THREAD

**Discussion by jbscherrer** [bonsai-rx/bonsai#1095] [Q&A] [answered]
Say I have some bonsai workflow that is dependent on a bunch of different packages, and I want to get a new computer running that workflow. Is there a way to automatically install all of the packages that the workflow is dependent on? Currently I just get a bunch of different error messages in series as I identify the nodes with missing dependencies and track down what package each of them needs, which isn't a very straightforward process. In other languages I would expect to be able to configure the package manager to download dependencies automatically. Is there a simple solution here I'm missing? Thanks!

**Comment by bruno-f-cruz** [comment, marked as answer]
Yes, there is a way to create an "environment". We will probably document this at some point, but check the attached folder for an example. This folder contains 4 important files:

- `Setup.ps1` and `Setup.cmd` are a set of script files that download a Bonsai release from the official GitHub page. You can change the target release by change the line: `https://github.com/bonsai-rx/bonsai/releases/download/2.7/Bonsai.zip` for a different release.

- `Nuget.config` keeps a record of what remote and local sources of NuGet packages should the package manager look for.

- `Bonsai.config` keeps a record of package dependencies.

The last two files always exist in the current Bonsai installation folder and are updated when a new package is added. As a result, to clone an environment, simply copy your local `.Config` files. As a bonus, as any text file, these can be easily versioned with Git. 

Finally, when it is time to deploy your solution, simply run the `Setup.cmd` and wait for the magic to happen. Bonsai will be downloaded and attempt to bootstrap all dependencies in the `Bonsai.config` file.

As we move forward with the documentation we will make an effort to include these little tips.

Hope it helps,
Bruno

[Bonsai.zip](https://github.com/bonsai-rx/bonsai/files/9833607/Bonsai.zip)

**Comment by jbscherrer** [comment]
This is very helpful, thanks Bruno! I'm really excited to see Bonsai getting more detailed documentation.
